Neos.ContentGraph.DoctrineDbalAdapter/src/DoctrineDbalContentGraphProjectionFactory.php

```diff
@@ -40,13 +41,16 @@ public function build(
             $projectionFactoryDependencies->contentRepositoryId
         );
 
+
+
         return new ContentGraphProjection(
             new DoctrineDbalContentGraphProjection(
                 $this->dbalClient,
                 new NodeFactory(
                     $projectionFactoryDependencies->contentRepositoryId,
                     $projectionFactoryDependencies->nodeTypeManager,
-                    $projectionFactoryDependencies->propertyConverter
+                    $projectionFactoryDependencies->propertyConverter,
+                    new DimensionSpacePointsRepository($this->dbalClient->getConnection(), $tableNamePrefix)
                 ),
```

Comment:
i just want to point out that the node factory was previously only relying on direct nodeRow input and did not trigger any db queries.

But as this is just a lazy (selfupdating) set of dimension space points and nothing too complex its fine?

I thought the idea of the `NodeFactory` was to be rather pure and simple.

Reply:
mmm, I guess we could move the dependency outwards to where the NodeFactory is used, seems a bit of a PITA, but maybe it's worth to keep the factory pure. I am fine either way.